stats/lib/platform-routes.js

```diff
@@ -14,39 +14,56 @@ import { filterPreHandlerHook, filterOnSendHook } from './request-helpers.js'
 
 /** @typedef {import('./typings.js').RequestWithFilter} RequestWithFilter */
 
-export const addPlatformRoutes = (app, pgPools) => {
+/**
+ * Create an adapter to convert Fastify pg to the expected pgPools format
+ * @param {any} pg Fastify pg object
+ * @returns {object} pgPools compatible object
+ */
+
+function adaptPgPools(pg) {
+  return {
+    stats: pg.stats,
+    evaluate: pg.evaluate,
+    end: async () => {} // Empty implementation
+  };
+}
+
+
+export const addPlatformRoutes = (app) => {
   app.register(async app => {
     app.addHook('preHandler', filterPreHandlerHook)
     app.addHook('onSend', filterOnSendHook)
 
     app.get('/stations/daily', async (/** @type {RequestWithFilter} */ request, reply) => {
-      reply.send(await fetchDailyStationCount(pgPools.evaluate, request.filter))
+      reply.send(await fetchDailyStationCount(request.server.pg.evaluate, request.filter))
     })
     app.get('/stations/monthly', async (/** @type {RequestWithFilter} */ request, reply) => {
-      reply.send(await fetchMonthlyStationCount(pgPools.evaluate, request.filter))
+      reply.send(await fetchMonthlyStationCount(request.server.pg.evaluate, request.filter))
     })
     app.get('/stations/desktop/daily', async (/** @type {RequestWithFilter} */ request, reply) => {
-      reply.send(await fetchDailyDesktopUsers(pgPools.stats, request.filter))
+      reply.send(await fetchDailyDesktopUsers(request.server.pg.stats, request.filter))
     })
     app.get('/measurements/daily', async (/** @type {RequestWithFilter} */ request, reply) => {
-      reply.send(await fetchDailyStationMeasurementCounts(pgPools.evaluate, request.filter))
+      reply.send(await fetchDailyStationMeasurementCounts(request.server.pg.evaluate, request.filter))
     })
     app.get('/participants/top-measurements', async (/** @type {RequestWithFilter} */ request, reply) => {
-      reply.send(await fetchParticipantsWithTopMeasurements(pgPools.evaluate, request.filter))
+      reply.send(await fetchParticipantsWithTopMeasurements(request.server.pg.evaluate, request.filter))
     })
     app.get('/participants/top-earning', async (/** @type {RequestWithFilter} */ request, reply) => {
-      reply.send(await fetchTopEarningParticipants(pgPools.stats, request.filter))
-    })
+      const pgPools = adaptPgPools(request.server.pg);
+      reply.send(await fetchTopEarningParticipants(pgPools.stats, request.filter))    })
```

Comment:
```suggestion
      reply.send(await fetchTopEarningParticipants(request.server.pg.stats, request.filter))    })
```

What do we need the `adaptPgPools` function for here?